modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/CrossProtocolLogoutTestCase.java

```diff
@@ -0,0 +1,411 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.identity.integration.test;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.oauth.stub.dto.OAuthConsumerAppDTO;
+import org.wso2.carbon.identity.sso.saml.stub.types.SAMLSSOServiceProviderDTO;
+import org.wso2.identity.integration.common.clients.oauth.OauthAdminClient;
+import org.wso2.identity.integration.common.clients.sso.saml.SAMLSSOConfigServiceClient;
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+import org.wso2.identity.integration.test.util.Utils;
+import org.wso2.identity.integration.test.utils.CommonConstants;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Integration test for cross protocol logout.
+ */
+public class CrossProtocolLogoutTestCase extends ISIntegrationTest {
+
+    private static final Log LOG = LogFactory.getLog(CrossProtocolLogoutTestCase.class);
+
+    private HttpClient client;
+    private List<NameValuePair> consentParameters = new ArrayList<>();
+    private static final String OIDC_APP_NAME = "playground2";
+    private static final String SAML_ISSUER = "travelocity.com";
+    private static String oidcAppClientId = "";
+    private OauthAdminClient adminClient;
+    private static final String ATTRIBUTE_CS_INDEX_VALUE = "1239245949";
+    private String sessionDataKeyConsent;
+    private String sessionDataKey;
+    private static final String SAML_SSO_LOGIN_URL =
+            "http://localhost:8490/travelocity.com/samlsso?SAML2.HTTPBinding=HTTP-POST";
+    private static final String ACS_URL = "http://localhost:8490/travelocity.com/home.jsp";
+    private static final String ACS_URL_ORI = "http://localhost:8490/%s/home.jsp";
+    private static final String COMMON_AUTH_URL = "https://localhost:9853/commonauth";
```

Comment:
Don't hardcode URLs, instead construct from backendURL in ISIntegrationTest

Reply:
Fixed.